File: backend/app/integrations/github/status_client.py
import requests

GITHUB_API = "https://api.github.com"

def get_commit_status(owner: str, repo: str, sha: str, token: str):
    url = f"{GITHUB_API}/repos/{owner}/{repo}/commits/{sha}/status"
    headers = {
        "Authorization": f"Bearer {token}",
        "Accept": "application/vnd.github+json"
    }
    return requests.get(url, headers=headers).json()


def get_check_runs(owner: str, repo: str, sha: str, token: str):
    url = f"{GITHUB_API}/repos/{owner}/{repo}/commits/{sha}/check-runs"
    headers = {
        "Authorization": f"Bearer {token}",
        "Accept": "application/vnd.github+json"
    }
    return requests.get(url, headers=headers).json()

File: backend/src/integrations/github/status-client.ts
import axios from 'axios';
import { GITHUB_DOTCOM_API } from './host.js';

function headersFor(token: string): Record<string, string> {
  return {
    'Authorization': `Bearer ${token}`,
    'Accept': 'application/vnd.github+json',
  };
}

export async function getCommitStatus(
  owner: string, repo: string, sha: string, token: string,
  apiBaseUrl: string = GITHUB_DOTCOM_API
): Promise<Record<string, unknown>> {
  const url = `${apiBaseUrl}/repos/${owner}/${repo}/commits/${sha}/status`;
  const r = await axios.get(url, { headers: headersFor(token) });
  return r.data;
}

export async function getCheckRuns(
  owner: string, repo: string, sha: string, token: string,
  apiBaseUrl: string = GITHUB_DOTCOM_API
): Promise<Record<string, unknown>> {
  const url = `${apiBaseUrl}/repos/${owner}/${repo}/commits/${sha}/check-runs`;
  const r = await axios.get(url, { headers: headersFor(token) });
  return r.data;
}
